src/fs.rs

```diff
@@ -136,6 +136,10 @@ impl FilesystemReader<'_> {
         let mut xattrs = BTreeMap::new();
 
         let names_size = listxattr(&filename, &mut [])?;
+        // See https://github.com/bytecodealliance/rustix/pull/1061#issuecomment-2518467534
+        // Basically current rustix 0.38 sometimes may expose the element type as a signed
+        // or unsigned variant, depending on the version of linux-raw-sys and rustix feature
+        // flags.
         let mut names = vec![0; names_size];
```

Comment:
A confusing thing here is that type inference is kicking in for the `0` here and we had conflicting requirements before in the wrong case. It'd help to be able to explicitly specify this array as the target type but I don't think rustix exposes that type alias right now even though it probably should because it's part of the public API.